### ISSUE TITLE
Unable to read blob data via database/sql API

### ISSUE THREAD

**Issue by d-smith** [rana/ora#120]
I am able to write blob data using the database/sql API, but attempts to read the data produces the following error:

<pre>
2016/10/11 11:26:27 sql: Scan error on column index 0: unsupported Scan, storing driver.Value type *ora.lobReader into type *[]uint8
</pre>

Below is code to reproduce this. The code assumes the following table definition:

<pre>
create table blob_sample (
    id integer,
    payload blob
);
</pre>

Best regards,

--doug

<pre>
package main

import (
    "database/sql"
    "encoding/json"
    "fmt"
    _ "gopkg.in/rana/ora.v3"
    "log"
    "time"
)

type ABC struct {
    A string
    B int
    C string
}

func fatal(err error) {
    if err != nil {
        log.Fatal(err)
    }
}

func main() {
    db, err := sql.Open("ora", "user/password@//localhost:1521/xe.oracle.docker")
    fatal(err)

    defer db.Close()

    err = db.Ping()
    fatal(err)

    id := time.Now().Unix()
    err = insertBlobData(db, id)
    fatal(err)

    abc, err := readBlobData(db, id)
    fatal(err)
    fmt.Printf("%v+v\n", abc)
}

func insertBlobData(db *sql.DB, id int64) error {
    abc := ABC{
        A: "a part", B: 42, C: "you see",
    }

    abcBytes, err := json.Marshal(&abc)
    if err != nil {
        return err
    }

    _, err = db.Exec("insert into blob_sample (id, payload) values (:1,:2)", id, abcBytes)
    return err
}

func readBlobData(db *sql.DB, id int64) (ABC, error) {
    var abc ABC

    rows, err := db.Query("select payload from blob_sample where id = :1", id)
    if err != nil {
        return abc, err
    }

    defer rows.Close()

    var abcBytes []byte

    for rows.Next() {
        err := rows.Scan(&abcBytes)
        if err != nil {
            return abc, err
        }
    }

    if err = rows.Err(); err != nil {
        return abc,err
    }

    err = json.Unmarshal(abcBytes, &abc)

    return abc, err
}
</pre>

**Comment by tgulacsi** [comment]
See #117.
TL;DR: use `ora.Cfg().Env.StmtCfg.Rset.SetClob(ora.S)` and `ora.Cfg().Env.StmtCfg.Rset.SetBlob(ora.R)` to instruct ora to read the LOB in full into a string/[]byte - by default, it returns an io.Reader.

**Comment by d-smith** [comment]
I tried adding the above lines both before and after db.Open, and still see the exact same error. Should the code above work with the v3 release or do I need a later commit?

<pre>
func main() {

	ora.Cfg().Env.StmtCfg.Rset.SetBlob(ora.B)
	ora.Cfg().Env.StmtCfg.Rset.SetClob(ora.S)

	db, err := sql.Open("ora", "user/password@//localhost:1521/xe.oracle.docker")
        etc...
</pre>

**Comment by tgulacsi** [comment]
Sorry, the correct constant is ora.Bin, not ora.B.

**Comment by tgulacsi** [comment]
See the extended TestLobSelect in `z_oracle_test.go` from 8707b22.

**Comment by d-smith** [comment]
I updated my sample to use ora.Bin and I still receive the same error. I am using the v3 release - do I need to grab the latest commits?

**Comment by tgulacsi** [comment]
At least a `go get -u gopkg.in/rana/ora.v3`.

Btw, @rana, I'm thinking about changing the defaults for Lobs to be ora.Bin and ora.S - it seems that everybody wants to slurp all the LOB into memory.
What's your opinion?

**Comment by d-smith** [comment]
Ok - go get -u sorted me out - I can now retrieve the blob data.

Thank you very much for your help.

--doug

**Comment by tgulacsi** [comment]
It's now in master, will be release with v3.8.
